.github/workflows/build-and-publish-master.yaml

```diff
@@ -24,6 +22,8 @@ jobs:
       - uses: codecov/codecov-action@v1
         with:
           fail_ci_if_error: true
+      - name: Fetch tag history
+        run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
```

Comment:
This is needed for the SNAPSHOT publish to compute the correct base version (without the tag history, it always pushes to `0.0.0-<stuff>-SNAPSHOT`)